mmv1/third_party/terraform/services/filestore/resource_filestore_instance_test.go

```diff
@@ -190,3 +190,92 @@ resource "google_filestore_instance" "instance" {
 }
 `, name)
 }
+
+func TestAccFilestoreInstance_deletionProtection_update(t *testing.T) {
+	t.Parallel()
+
+	name := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+	location := "us-central1-a"
+	tier := "ZONAL"
+
+	deletionProtection := true
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckFilestoreInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFilestoreInstance_deletionProtection_create(name, location, tier),
+			},
+			{
+				ResourceName:            "google_filestore_instance.instance",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"zone", "location"},
+			},
+			{
+				Config: testAccFilestoreInstance_deletionProtection_update(name, location, tier, deletionProtection),
+			},
```

Comment:
```suggestion
			},
			{
				ResourceName:            "google_filestore_instance.instance",
				ImportState:             true,
				ImportStateVerify:       true,
				ImportStateVerifyIgnore: []string{"zone", "location"},
			},
```

We should generally use `ImportState` steps between every step unless there's a specific reason we're avoiding doing so!

Reply:
Thanks for the input. Can I please ask why is that so? It seems we don't test the state itself rather the fact that it exists so I don't fully understand why this is done.

Reply:
`ImportStateVerify` does a full deep comparison of the state after the apply and a fresh state created during import. It catches pretty much all non-adversarial issues with reading or setting values (for example, hardcoding a specific value wouldn't cause an issue- but would be caught during review).

In the past we wrote "TestCheck" checks for specific fields, but they tended to be brittle / took extra code to write (in particular, needing a distinct API fetch per resource, including a fair bit of flexibility in auth). Import-driven checks superseded them.